Add aws_sagemaker_mlflow_tracking_server resource

Ezzmo opened the pull request:
### Description of your changes

This PR adds the `mlflowtrackingservers.sagemaker.aws.upbound.io` resource

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

I tested locally with uptest. The MlFlow tracking server created successfully and tasks succeeded

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->

[contribution process]: https://git.io/fj2m9

Ezzmo commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

Ezzmo commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

turkenf commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

Ezzmo commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

Ezzmo commented:
@turkenf do you mind running the tests again? The error may have been transient, doesn't look code related

jeanduplessis commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

jeanduplessis commented:
@Ezzmo It looks like the provider pod logs have some info on errors

[provider-aws-sagemaker logs](https://github.com/user-attachments/files/17777867/logs.txt)
[provider-aws-iam logs](https://github.com/user-attachments/files/17777868/logs.txt)

turkenf reviewed (commented):
Thank you for your contribution @Ezzmo. When we look at the uptest [logs](https://github.com/crossplane-contrib/provider-upjet-aws/actions/runs/11859182412/job/33051544694), we see that the resources cannot resolve the references. I left comments about this for you to consider.

Ezzmo commented:
Thanks @turkenf , I missed those (oops)

I've updated the references for the differences resources and fixed the S3 reference, and some metadata mistakes

Ezzmo commented:
@turkenf Looks like it needs another run

turkenf commented on config/externalname.go:
```diff
@@ -2281,6 +2281,8 @@ var TerraformPluginSDKExternalNameConfigs = map[string]config.ExternalName{
 	// SageMaker Code Images can be imported using the name
 	"aws_sagemaker_image_version": config.IdentifierFromProvider,
+	// Sagemaker MLFlow tracking server can be imported using the name
+	"aws_sagemaker_mlflow_tracking_server": config.NameAsIdentifier,
```
To use `NameAsIdentifier` as an external name, the resource must be imported with the name, and the name argument must be present. See https://github.com/crossplane/upjet/blob/main/docs/adding-new-resource.md#case-1-name-as-identifier

When I look at the resource's [import document](https://registry.terraform.io/providers/hashicorp/aws/5.73.0/docs/resources/sagemaker_mlflow_tracking_server#import), it says that it can be imported with `workteam_name`. I assume there is a mistake here and that is because I don't see an argument called `workteam_name`. Instead, `tracking_server_name` can be used, and in this case, the most appropriate external name configuration is [ParameterAsIdentifier](https://github.com/crossplane/upjet/blob/main/docs/adding-new-resource.md#case-2-parameter-as-identifier).

Could you please regenerate the resource with the following configuration?

```suggestion
	"aws_sagemaker_mlflow_tracking_server": config.ParameterAsIdentifier("tracking_server_name"),
```

Ezzmo replied:
I made the mistake of thinking `id` was acceptable for `NameAsIdentifier`

Thanks @turkenf. I've updated

turkenf commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

turkenf commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

Ezzmo commented:
@turkenf Thanks for helping in the Slack channel. Tests ran successfully finally after increasing the timeout.

```
--- PASS: kuttl (2843.65s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (2842.41s)
```

Hopefully will run successfully here

turkenf commented:
/test-examples="examples/sagemaker/v1beta1/mlflowtrackingserver.yaml"

Ezzmo commented:
@turkenf Rebasing this is proving very hairy, I'm going to produce a new branch with a new PR with a single commit to clean this all up. I suspect the way I updated my feature branch has caused problems now.
